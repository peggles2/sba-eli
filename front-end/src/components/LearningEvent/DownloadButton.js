import React from "react";
import { Button, Icon } from "semantic-ui-react";

const DownloadButton = props => {
  return (
    <Button
      icon
      as="a"
      href={props.url}
      aria-label="download"
      download
      className="mobile hidden"
      floated="right"
    >
      <Icon name="download" />
    </Button>
  );
};

export default DownloadButton;
