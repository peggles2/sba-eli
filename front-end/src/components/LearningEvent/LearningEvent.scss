@import "../../assets/styles/base";
/* Prototyping for infographic until more robust css in place */
.learning-event-container {
  border: 1px solid rgba(34, 36, 38, 0.15);
  box-shadow: 0 1px 2px 0 rgba(34,36,38, 0.15);
  padding: 15px 0px 65px 0px;
}

.learning-event-header {
  padding-top: 30px;
  padding-left: 45px;
  padding-right: 45px;
}

.learning-event-header-video {
  padding-top: 10px;
  padding-bottom: 20px;
  padding-left: 45px;
  padding-right: 45px;
}

.learning-event-p {
  font-size: 1.3em;
}

.learning-event-footer { 
  padding-top: 30px;
  padding-left: 45px;
  padding-right: 45px;
}

.infographic-header-footer {
  background-color: #0b2f6b;
}

.header-text {
  font-size: 40px;
}

.row.infographic-header-border {
  padding-top: 0px !important;
  background-color: #a8e7ee;
}
.row.infographic-header-border > img {
  margin-top: -1px;
}

.skyblue-white-gradient {
  background-image: linear-gradient(#a8e7ee, #ffffff);
}
.no-padding {
  padding: 0;
}
.largest-text {
  font-size: 100px;
  font-weight: bold;
}

.small-text {
  font-size: 25px;
}

.medium-text {
  font-size: 40px;
  font-weight: bolder;
}
