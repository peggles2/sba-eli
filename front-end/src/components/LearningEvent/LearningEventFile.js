import React, { Component } from "react";
import LearningEventVideo from "./LearningEventVideo"
import axios from "axios";

export default class LearningEventFile extends Component {
  state = {
    fileInfo: []
  };

  componentDidMount() {
    this.getFileUrl();
  };

  getFileUrl() {
    const url = this.props.event.url;

    axios
      .get(url)
      .then(res => {
        const fileInfo = res.data;
        console.log(fileInfo);
        this.setState({ fileInfo });
      })
      .catch(error => {
        console.log(error);
      });
  };

  renderFileByType() {
    const file = this.state.fileInfo;

    if (file.mime_class === "video") {
      return <LearningEventVideo file={file} />;
    } else {
      return <LearningEventVideo />;
    }
  };

  render() {
    return  this.renderFileByType();
  }
}
