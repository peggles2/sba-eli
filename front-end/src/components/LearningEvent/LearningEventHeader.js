import React from "react";
import { Header } from "semantic-ui-react";

const LearningEventHeader = (props) => {
  const event = props.event;

  return(
    <div className="learning-event-header">
      <Header as='h1'>{event.title}</Header>
    </div>
  )
}

export default LearningEventHeader;
