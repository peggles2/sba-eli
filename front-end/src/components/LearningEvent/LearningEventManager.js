import React, { Component } from "react";
import mime from "mime-types";
import LearningEventInfoCard from "./LearningEventInfoCard";
import LearningEventPage from "./LearningEventPage";
import LearningEventVideo from "./LearningEventVideo";

export default class LearningEventManager extends Component {
  eventFileManager(event) {
    const fileType = event.eventContent.mime_class;
    const url = event.eventContent.url;

    switch (fileType) {
      case 'video':
        return <LearningEventVideo url={url} event={event} />;
      default:
        return this.infoCard(event)
    }
  }

  eventURLManager(event) {
    const url = event.external_url;
    const mimeType = mime.lookup(url).split("/")[0];

    switch(mimeType) {
      case 'video': 
        return <LearningEventVideo url={url} event={event}/>;
      default:
        return this.infoCard(event);
    }
  }

  infoCard(event) {
    return <LearningEventInfoCard event={event} />;
  }

  renderEventContentByType() {
    const event = this.props.event;

    switch(event.type) {
      case 'Page':
        return <LearningEventPage event={event} />;
      case 'File':
        return this.eventFileManager(event);
      case 'ExternalUrl':
        return this.eventURLManager(event);
      default:
        return this.infoCard(event);
    }
  }

  render() {
    return this.renderEventContentByType();
  }
}
