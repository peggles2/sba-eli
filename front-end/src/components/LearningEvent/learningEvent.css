/* Prototyping for infographic until more robust css in place */
.infographic {
  font-family: "Source Sans Pro", sans-serif;
}

.sba-yellow {
  color: #ffab3b;
}

.sba-white {
  color: #ffffff;
}

.sba-blue {
  color: #0b2f6b;
}

.infographic-header-footer {
  background-color: #0b2f6b;
}

.header-text {
  font-size: 40px;
}

.row.infographic-header-border {
  padding-top: 0px !important;
  background-color: #a8e7ee;
}
.row.infographic-header-border > img {
  margin-top: -1px;
}

.skyblue-white-gradient {
  background-image: linear-gradient(#a8e7ee, #ffffff);
}
.no-padding {
  padding: 0;
}
.largest-text {
  font-size: 100px;
  font-weight: bold;
}

.small-text {
  font-size: 25px;
}

.medium-text {
  font-size: 40px;
  font-weight: bolder;
}
