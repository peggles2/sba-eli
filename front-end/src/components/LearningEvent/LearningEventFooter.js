import React, { Component } from "react";
import { Link } from "react-router-dom";
import { Icon, Button } from "semantic-ui-react";
import LearningEvent from "./LearningEvent";

export default class LearningEventFooter extends Component {
  parseNextPrev(modules, event) {
    let paging = { next: null, prev: null };

    if (event && event.position && modules) {
      const pos = event.position - 1; //To account for zero based array indexes
      paging.next = modules[pos + 1]
        ? LearningEvent.getEventPath(
            this.props.courseId,
            event.module_id,
            modules[pos + 1].id
          )
        : null;
      paging.prev = modules[pos - 1]
        ? LearningEvent.getEventPath(
            this.props.courseId,
            event.module_id,
            modules[pos - 1].id
          )
        : null;
    }
    return paging;
  }

  render() {
    const np = this.parseNextPrev(this.props.module, this.props.event);

    return (
      <div className="page-navigation-buttons">
        {np.prev && (
          <Link to={np.prev}>
            <Button icon labelPosition="left">
              <Icon name="left arrow" />
              Previous
            </Button>
          </Link>
        )}
        {np.next && (
          <Link to={np.next}>
            <Button icon labelPosition="right" primary>
              Next
              <Icon name="right arrow" />
            </Button>
          </Link>
        )}
      </div>
    );
  }
}
