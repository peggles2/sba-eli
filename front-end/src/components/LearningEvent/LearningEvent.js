import React, { Component } from "react";
import { Container } from "semantic-ui-react";
import axios from "axios";
import LearningEventHeader from "./LearningEventHeader"
import LearningEventManager from "./LearningEventManager";
import LearningEventFooter from "./LearningEventFooter";
import { Header, Container, Divider } from "semantic-ui-react";
import LearningEventFooter from "./LearningEventFooter";
import "./LearningEvent.scss";
import MetaTags from '../SEO/MetaTags'

export default class LearningEvent extends Component {
  constructor(props) {
    super(props);

    this.state = {
      learningEvent: {}
    };
  }

  componentDidMount() {
    this.setLearningEvent();
  }

  componentWillReceiveProps(newProps) {
    //Added this to account for react routing not updating with different props
    if (this.props !== newProps) {
      this.props = newProps;
      this.setLearningEvent();
    }
  }

  setLearningEvent() {
    const { course_id, module_id, id: event_id } = this.props.match.params;
    const url =
      process.env.REACT_APP_SERVICE_HOST + `/learning_events/${event_id}`;
    const url2 =
      process.env.REACT_APP_SERVICE_HOST + `/learning_events/`;

    const eventParams = {
      course_id,
      module_id
    };

    axios
      .get(url, { params: eventParams })
      .then(res => {
        const learningEvent = res.data;
        this.setState({ learningEvent });
      })
      .catch(error => {
        console.log(error);
      });

    axios
      .get(url2, { params: eventParams })
      .then(res => {
        const learningModule = res.data;
        this.setState({ learningModule });
      })
      .catch(error => {
        console.log(error);
      });
  }

  render() {
    const event = this.state.learningEvent;

    return (
      <Container className="learning-event-container">
        <MetaTags metaTitle={event.title}
                metaDescription={event.description}
                canonicalUrl=""/>
        <LearningEventHeader event={event} />
        <LearningEventManager event={event} />
        <Divider />
        <LearningEventFooter courseId={this.props.match.params.course_id} module={this.state.learningModule} event={this.state.learningEvent}/>
      </Container>
    );
  }
}
