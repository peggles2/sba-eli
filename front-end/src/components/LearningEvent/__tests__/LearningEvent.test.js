import React from "react";

import { LearningEvent } from "../LearningEvent";
import LearningEventManager from "../LearningEventManager";
import MetaTags from "../../SEO/MetaTags";
import { shallow } from "enzyme";
import LearningEventHeader from "../LearningEventHeader";
import { Helmet } from "react-helmet";
import LearningEventFooter from "../LearningEventFooter";

function eventWrapper() {
  const match = { params: { course_id: 1, module_id: 1, id: 1 } };
  const title = "learning event title";
  const desc = "learning event description";
  const learningEvent = { title: title, description: desc };
  const props = {
    learningEvent,
    match,
    learningEventsCollection: {},

    dispatch: () => {}
  };

  const wrapper = shallow(<LearningEvent {...props} />);

  return { title, desc, wrapper };
}

describe("LearningEvent", () => {
  it("should render meta tags with Learning Event specific information", () => {
    const { title, desc, wrapper } = eventWrapper();

    expect(wrapper.find(MetaTags).exists()).toBe(true);
    const metaWrapper = wrapper.find(MetaTags).dive();
    const helmetProps = metaWrapper.find(Helmet).props();

    expect(helmetProps.title).toBe(title);
    expect(helmetProps.meta[0].name).toBe("description");
    expect(helmetProps.meta[0].content).toBe(desc);

    //TODO: add Canonical URL validation when we have final urls
    //expect(helmetProps.link[0].rel).toBe("canonical");
    //expect(helmetProps.link[0].href).toBe("https://sba.gov/eli");
  });

  it("should render a LearningEventHeader", () => {
    const { wrapper } = eventWrapper();

    expect(wrapper.find(LearningEventHeader).exists()).toBe(true);
  });

  it("should render a LearningEventFooter", () => {
    const { wrapper } = eventWrapper();

    expect(wrapper.find(LearningEventFooter).exists()).toBe(true);
  });

  it("should render a <LearningEventManager>", () => {
    const { wrapper } = eventWrapper();

    expect(wrapper.find(LearningEventManager).exists()).toBe(true);
  });
});
