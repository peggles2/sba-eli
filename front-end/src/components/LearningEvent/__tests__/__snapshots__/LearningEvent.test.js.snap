// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningEvent should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LearningEvent
    match={
      Object {
        "params": Object {
          "course_id": 1,
          "id": 1,
          "module_id": 1,
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header
          as="h1"
        />,
        <LearningEventManager
          event={Object {}}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "h1",
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "event": Object {},
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header
            as="h1"
          />,
          <LearningEventManager
            event={Object {}}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "event": Object {},
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
